### PR TITLE
Add Publish Endpoint & Option to sui-move to print hex

### PR DESCRIPTION
**Pull request by arun-koshy**
Usage:
```
sui-move --hex --path sui_programmability/examples/fungible_tokens build
   Compiling sui-framework v0.1.0 (/Users/arunkoshy/projects/mystenlabs/sui/sui_programmability/framework)
   Compiling sui-adapter v0.1.0 (/Users/arunkoshy/projects/mystenlabs/sui/sui_programmability/adapter)
   Compiling sui_core v0.1.0 (/Users/arunkoshy/projects/mystenlabs/sui/sui_core)
   Compiling sui v0.1.0 (/Users/arunkoshy/projects/mystenlabs/sui/sui)
    Finished dev [unoptimized + debuginfo] target(s) in 11.97s
     Running `target/debug/sui-move --hex --path sui_programmability/examples/fungible_tokens build`
["a11ceb0b0400000009010008020814031c3704530a055d7207cf017408c302280aeb02050cf0024200000101010201030000020001040c01000101010c010001030302000005000100000602010000070304000008050100010507010100010a090a0102030b0b0c00020c0d01010801070e0f01000108100101000406050607080806090603070b010108000b02010800070803000107080303070b0101080003070803010b02010800030b0101080005070803010800020b02010900070b01010900010b01010800020900070803010b01010900010608030105020900050303070b01010900070803010b02010900020b0101090005074d414e4147454404436f696e085472616e73666572095478436f6e746578740b5472656173757279436170046275726e04696e6974046d696e740c7472616e736665725f6361700b64756d6d795f6669656c640f6372656174655f63757272656e63790673656e646572087472616e736665720000000000000000000000000000000000000000000000000000000000000000000000000000000200020109010001000001040b010b0038000201000000080b0912000a0038010c010b010b002e11063802020201000001050b010b000b023803020301000001040b000b0138040200"]
Build Successful
```

`Publish Endpoint`: It will perform proper verification and linking to make sure the package is valid. If some modules have initializers, these initializers will also be executed in Move (which means new Move objects can be created in the process of publishing a Move package). Gas budget is required because of the need to execute module initializers.

Usage:
```
curl --location --request POST 'http://127.0.0.1:5001/publish' \
--header 'Content-Type: application/json' \
--data-raw '{
    "sender": "81E5042AB348B7E325DAECAE8BC099588FCF4192",
    "compiledModules": ["a11ceb0b0400000009010008020814031c3704530a055d7207cf017408c302280aeb02050cf0024200000101010201030000020001040c01000101010c010001030302000005000100000602010000070304000008050100010507010100010a090a0102030b0b0c00020c0d01010801070e0f01000108100101000406050607080806090603070b010108000b02010800070803000107080303070b0101080003070803010b02010800030b0101080005070803010800020b02010900070b01010900010b01010800020900070803010b01010900010608030105020900050303070b01010900070803010b02010900020b0101090005074d414e4147454404436f696e085472616e73666572095478436f6e746578740b5472656173757279436170046275726e04696e6974046d696e740c7472616e736665725f6361700b64756d6d795f6669656c640f6372656174655f63757272656e63790673656e646572087472616e736665720000000000000000000000000000000000000000000000000000000000000000000000000000000200020109010001000001040b010b0038000201000000080b0912000a0038010c010b010b002e11063802020201000001050b010b000b023803020301000001040b000b0138040200"],
    "gasObjectId": "10A1EA435F29F11392C385BDCE4CB8BD69636B9E",
    "gasBudget": 10000
}'
```

Output:
```
{
    "certificate": {
        "signedAuthorities": [
            "7e645059dbd3f91a61e3fb8cbce8a3fa6f8aba73ae8e6eb6c359adb09b69c8e9",
            "767553ce3e105ba905d1d2c69acd697c167abc7935d53bc92a6337f5502d8e55",
            "6b16c80392acb302274df2bc6458be7a1f731a264ef0bdb9842ebc1bf0614644"
        ]
    },
    "publishResults": {
        "createdObjects": [
            {
                "id": "10C98C65DEF2C8B3D5F39C8A8A99122D79417E27",
                "obj_type": "0x2::Coin::TreasuryCap<0x7e5f0d93b3030408abcb6042a47d3113097bcbdb::MANAGED::MANAGED>",
                "owner": "AddressOwner(k#003f443669c425ba9a0cbfeb9f47ac540dc591ad)",
                "readonly": "false",
                "version": "1"
            }
        ],
        "package": {
            "object_digest": "o#fe0b97ff41ba14beb5a9f2f9451dc6d36c80431f6386b368a93e14aba1d66bd8",
            "object_id": "7E5F0D93B3030408ABCB6042A47D3113097BCBDB",
            "version": 1
        },
        "updatedGas": {
            "id": {
                "id": {
                    "id": {
                        "bytes": "0afa9742ed761296a28ef5c421f61988d213aa7a"
                    }
                },
                "version": 4
            },
            "value": 99371
        }
    }
}
```

### CODE REVIEW

File: sui/src/sui-move.rs [hunk 1]
```diff
@@ -58,6 +75,9 @@ struct MoveOpt {
     /// Whether we are building/testing the std/framework code.
     #[structopt(long)]
     std: bool,
+    /// Whether we are printing in hex.
+    #[structopt(long)]
+    hex: bool,
```
Comment (lxfind) [inline sui/src/sui-move.rs:80]: I think this should be a sub-option of `Build` command.

File: sui/src/sui-move.rs [hunk 2]
```diff
@@ -35,6 +45,13 @@ impl MoveCommands {
         Ok(())
     }
 
+    fn print_hex(path: &Path, is_std_framework: bool) -> Result<Vec<String>, anyhow::Error> {
```
Comment (lxfind) [inline sui/src/sui-move.rs:48]: This function seems simple enough that it probably doesn't deserve to be a separate function

**Review by lxfind — COMMENTED** [review]
Thanks! Some suggestions commented

**Comment by arun-koshy** [comment]
Updated usage:

```
sui-move --path sui_programmability/examples/games/hero build --hex
    Finished dev [unoptimized + debuginfo] target(s) in 0.44s
     Running `target/debug/sui-move --path sui_programmability/examples/games/hero build --hex`
["a11ceb0b040000000b01001202123403469a0104e0011a05fa01d60107d003e40308b4073c06f007700ae008410ca1098f050db00e1000000101020202030204020502060207020800090800000a0300000b080000000c00000c0c00000d0c0002020c0100010404020008080200010107010000051e040005050700000e000100000f0102000010030100001104050000120006000013070800001409010000150a0b0000160c010000170d010000180e060000190f0100001a100100001b110100001c120b00082a140200072b15010108082c0c1600012d17180100022e1b0b0100062f190b0001301d180100053116010001321f20010001331f2101000134241701000135242801000536292a0003371701010310051206131a101c15061706180610231906102510261a061c2b020b060108070708080001050306080303070808020805070808010803010805020708030805010b090108050207080308040106080301030107080802070805030107080305070802030305070808040305070802070808030708030800070808010608050208030805010608080209000501080a010900010b0901090002030301080701060b06010900010b0601080702070b09010900090003080a030301060b0901090001010106090002050501080201070b090109000108000108040603080a03030303010709000106080a0106080b010801044865726f064f7074696f6e04436f696e054576656e7403474153024944044d617468085472616e73666572095478436f6e7465787404426f61720e426f6172536c61696e4576656e740947616d6541646d696e06506f74696f6e0553776f72640c616371756972655f6865726f0561646d696e146173736572745f6865726f5f737472656e6774680b6372656174655f6865726f0c6372656174655f73776f72640b65717569705f73776f7264046865616c0d6865726f5f737472656e67746804696e69740e6c6576656c5f75705f73776f72640c72656d6f76655f73776f72640973656e645f626f61720b73656e645f706f74696f6e04736c61790e73776f72645f737472656e6774680269640b56657273696f6e6564494402687008737472656e6774680e736c617965725f61646472657373046865726f04626f61720d626f6172735f637265617465640f706f74696f6e735f637265617465640a657870657269656e63650573776f726407706f74656e6379056d616769630673656e646572087472616e73666572066e65775f696404736f6d650576616c7565036d696e0c737761705f6f725f66696c6c0664656c6574650769735f736f6d6506626f72726f7707657874726163740a626f72726f775f6d757405696e6e657204656d69740000000000000000000000000000000000000000000000000000000000000000000000000000000100000000000000000000000000000000000000020514a5e6dbcf33730ace6ec8b400ff4788c1f150ff7e0308050000000000000003080000000000000000030801000000000000000308030000000000000003080200000000000000030804000000000000000308e80300000000000003080a00000000000000030864000000000000000002031d080a1f032003010203210522080b23080b0202031d080a240325030302041d080a1f032603270b090108050402021d080a28030502031d080a2903200300010000130e0b000a0111040c030b030a0111030c020b020b012e110f3800020100000001020700020201000001080b0011070b01210307070127020301000001080b0111110664000000000000000600000000000000000b00380112030204010000191c0e0038020c030a03070926030b0b01010704270b00110138030b0307091707091a0c020b0111110b02070811140601000000000000001205020501000001050b000f000b01380402060100001e130b0113040c040c020b0211160a001001140b04160c030b03070711140b000f0115020701000019240a00100114060000000000000000210307050b0b0001060000000000000000020a0010003805031005160a0010003806110e0c0105180600000000000000000c010b010c020a001002140b00100114180b02160208000000221611010c020a002e110f0c010e010e0221030e0b00010705270b00111106000000000000000006000000000000000012020b023807020900000001090a001003140b01160b000f0315020a010000010c0a001000380503080b00010706270b000f003808020b01000001100a00100414060100000000000000160b000f04150b0411110b010b0212000b033809020c010000010f0a02100514060100000000000000160b020f05150b0311110b0012040b01380a020d010000274c0b0113000c050c080c040a002e11070c070a080c030a001001140c060a030a0724031405270b030a07170c030a060a052603220b00010b02010702270b060a05170c06050f0b060a000f01150a001002140b08160a000f02150a00100038050338053d0a000f00380b06010000000000000011090b022e110f0b001006111b140e04111b141201380c0b041116020e01000001080a001007140b0010031416020303030103020502020102020300050100", "a11ceb0b040000000b01000c020c2403304b047b0c0587017c078302ba0208bd042806e5041e0a8305230ca605a7020dcd070a000000010102010301040105000602000007080000080c000505020002020c01000101010c000310040002120c0100010009000100000a020100000b030400000c050600000d07010002170a04010005180b0c0002190d0e0100041a10010108021b11120102051c131400021d16040100031e0c0100011f18040005090709080f090908150b0904070801030507080300030303070803010608020103020608050802010b0401080002080205030308020301080001060b07010900010708030108060303070b07010900070803010b0401090001080202090005020900070803010b0701090001060803010501080101060b040109000208060b0401080001060805075365614865726f044865726f04436f696e024944085472616e73666572095478436f6e746578740352554d0c5365614865726f41646d696e0a5365614d6f6e737465720e6372656174655f6d6f6e7374657204696e69740e6d6f6e737465725f72657761726404736c6179107472616e736665725f6d6f6e737465720b64756d6d795f6669656c640269640b56657273696f6e656449440c74726561737572795f6361700b5472656173757279436170106d6f6e73746572735f6372656174656410746f6b656e5f737570706c795f6d61780b6d6f6e737465725f6d6178067265776172640c746f74616c5f737570706c79066e65775f6964046d696e74087472616e736665720f6372656174655f63757272656e63790673656e6465720576616c75650664656c6574650d6865726f5f737472656e677468000000000000000000000000000000000000000000000000000000000000000000000000000000020308000000000000000003080200000000000000030801000000000000000002010e010102050f0806110b070108001303140315030202020f0806160b040108000001000008430a00100038000c040a001001140c060a010a062303120b03010b0001060000000000000000270b060a01170b0426031e0b03010b0001060100000000000000270a00100214060100000000000000170a0010031426032e0b03010b0001060200000000000000270a0311060b010a000f000b03380112020c050a00100314060100000000000000160b000f03150b050b0238020201000000011f0a000600000000000000002403080b02010702270a010600000000000000002403100b02010701270a0211060912000a0238030600000000000000000b000b0112010b022e110a3804020201000001040b0010043805020301000017100b0113020c030c020b02110c0b00110d0e03380526030e0700270b03020401000001040b000b013802020101010301040102020100", "a11ceb0b040000000b01000e020e1e032c3e046a0605706b07db01fe0108d903280681040a0a8b040d0c980489010da1050400000001000201030104010501060007080002080c000606020003030c010001020c020001010c00040f04000009000100000a020300000b040100000d0506000213070300061408090006150a0b0005160c010108041709010002180e0100020d10060003191213010007040b110706040801030507080200010608000103010800030608050800070802010b030108040106080101070802010806010608020105020900050308060801050208010506030b0301080408060801050b0301080402060805080101080403070b0301090003070802010b030109000d5365614865726f48656c706572044865726f075365614865726f04436f696e024944085472616e73666572095478436f6e746578741548656c704d65536c6179546869734d6f6e737465720a5365614d6f6e73746572066372656174650c6f776e65725f7265776172640f72657475726e5f746f5f6f776e65720352554d04736c61790269640b56657273696f6e65644944076d6f6e737465720d6d6f6e737465725f6f776e65720d68656c7065725f7265776172640e6d6f6e737465725f726577617264066e65775f69640673656e646572087472616e736665720664656c657465107472616e736665725f6d6f6e7374657208776974686472617700000000000000000000000000000000000000000000000000000000000000000000000000000002030800000000000000000002040e0806100801110512030001000001140e0011040a012403090b03010700270a0311050b000b032e11060b0112000b023800020101000001080a00100011040b001001141702020100000d0c0b001300010c030c020c010b0111080b020b03110902030100000f160b0113000c030c070c060c050b0511080b000b06110a0c080d080b030b0238010c040b080b0738020b04020001000300"]
Build Successful
```

File: sui/src/sui-move.rs [hunk 3]
```diff
@@ -12,7 +12,11 @@ use structopt::StructOpt;
 pub enum MoveCommands {
     /// Build and verify Move project
     #[structopt(name = "build")]
-    Build,
+    Build {
+        /// Whether we are printing in hex.
+        #[structopt(long)]
+        hex: bool,
```
Comment (lxfind) [inline sui/src/sui-move.rs:18]: I recommend renaming this to make it more explicit. Something like dump_bytecode_as_hex

**Review by lxfind — APPROVED** [review]

**Comment by arun-koshy** [comment]
Updated usage: 

`sui-move --path sui_programmability/examples/games/hero build --dump-bytecode-as-hex`

**Pull request merged**